napari/_app_model/context/_layerlist_context.py

```diff
@@ -134,11 +149,11 @@ def _empty_shapes_layer_selected(s: LayerSel) -> bool:
     return any(x._type_string == "shapes" and not len(x.data) for x in s)
 
 
-class LayerListContextKeys(ContextNamespace['LayerSel']):
-    """These are the available context keys relating to a LayerList.
+class LayerListSelectionContextKeys(ContextNamespace['LayerSel']):
+    """Available context keys relating to the selection in a LayerList.
 
-    along with default value, a description, and a function to retrieve the
-    current value from layers.selection
+    Consists of a default value, a description, and a function to retrieve the
+    current value from `layers.selection`.
```

Comment:
Since it's not meant to be used or known about by non-napari internal projects, I don't think that label is necessary here and may add noise to the future release notes.

Reply:
Is this 'breaking change' marker aimed at plugin developers, or dependencies like npe2 (where there is a lot of overlap with napari developers)?

Reply:
I'm not sure exactly @napari/core-devs what are your thoughts? Personally, I view npe2 as internal, treated as if it is an extension of the main napari repo for things like protected functions.

As far as plugin developers, this wouldn't be a breaking change as long as they stick to our public API.

Breaking changes are mainly aimed at people who externally interact with our API since they require a deprecation notice.

Reply:
Do you know about the usage of its outside napari repository? I do not see the conditional option in plugin contributions (based on docs).

Reply:
>  aimed at people who externally interact with our API since they require a deprecation notice.

Sorry my question was more, apart from plugin devs, does anyone else external interact with our API?

> Do you know about the usage of its outside napari repository? I do not see the conditional option in plugin contributions (based on docs).

No but it is planned to be made public as part of nap 6 (#6174)

Reply:
Given that this is in a private file and not exposed to plugin developers yet, I have no qualms about changing the name without a deprecation strategy here. A code search for `LayerListContextKeys` only revealed napari forks - although @Czaki there were a couple of mentions in `napari-release-tools` but I think they're just docs?

Reply:
> Sorry my question was more, apart from plugin devs, who else external interacts with our API?

In theory, any napari user **could** be writing scripts that make use of these context keys in some way, including making on the fly contributions, but it's exceedingly unlikely that they're doing it right now.